### PR TITLE
Section 4: Finish ChatEntity

### PR DESCRIPTION
**Pull request by thompsop1sou**
This PR reworks previous section 4a, "Creating ChatEntity", into section 4, "ChatEntity Conversations". Section 4 will now describe how to expand upon the ChatEntity and ChatAI scripts that were created in section 2, "Simple Chat NPCs".

### CODE REVIEW

**Comment by thompsop1sou** [comment]
@defreez This PR should be ready for review.

Would you be able to look particularly at the text for the `ChatDescr` property for Gralk? I wasn't sure if this is what we should use at this point, since we haven't set up the guess evaluation yet.

**Comment by defreez** [comment]
I think it is fine to use the full Gralk ChatDescr at this point. Without the guess evaluation at this point Gralk cannot move out of the way automatically, but that is expected and we have the ALT key workaround.

File: docs/first_semantic_game/04.chatEntity_conversations.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    [GlobalClass]
+    public partial class ChatEntity : Area2D
+    {
+        (...)
+    }
+
+.. note::
+
+    Changing ``ChatEntity`` to inherit from ``Area2D`` may break the ``ArguingTrolls`` scene
+    created in sections :doc:`01.project_setup` and :doc:`02.simple_chat_npcs`.
```
Comment (defreez) [inline docs/first_semantic_game/04.chatEntity_conversations.rst:30]:
```suggestion
.. note::

    Changing ``ChatEntity`` to inherit from ``Area2D`` will break the ``ArguingTrolls`` scene
    created in sections :doc:`01.project_setup` and :doc:`02.simple_chat_npcs`. This is expected.
    You may delete the ``ArguingTrolls`` scene at this point.
```

File: docs/first_semantic_game/04.chatEntity_conversations.rst [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+Now that ``ChatEntity`` inherits from ``Area2D``, we can write some properties and methods to keep
+track of other nearby entities. This will be used to display a list of entities to the player and
+will allow them to start a conversation with the nearest entity.
+
+.. tip::
+
+    We won't set up the NPCs to use this feature in *this* project, but it could be useful if, for
+    example, you want GPT-4 powered NPCs to be able to see and keep track of objects in the world
+    around them.
```
Comment (defreez) [inline docs/first_semantic_game/04.chatEntity_conversations.rst:43]:
```suggestion
.. tip::

    Another use of the nearby entity list might be to give AI-powered NPCs the ability to see and keep track of objects in the world around them.
```

**Review by defreez — COMMENTED** [review]
LGTM

I was able to follow along without any trouble.

**Review by defreez — APPROVED** [review]

**Pull request merged**